src/database/DataTypes/Transaction.js

```diff
@@ -157,25 +197,26 @@ export class Transaction extends Realm.Object {
   confirm(database) {
     if (this.isConfirmed) throw new Error('Cannot confirm as transaction is already confirmed');
     if (this.isFinalised) throw new Error('Cannot confirm as transaction is already finalised');
-    this.items.forEach((transactionItem) => {
-      transactionItem.batches.forEach((transactionBatch) => {
-        const itemBatch = transactionBatch.itemBatch;
-        const newNumberOfPacks = this.isIncoming ?
-          itemBatch.numberOfPacks + transactionBatch.numberOfPacks :
-          itemBatch.numberOfPacks - transactionBatch.numberOfPacks;
-        itemBatch.packSize = transactionBatch.packSize;
-        itemBatch.numberOfPacks = newNumberOfPacks;
-        itemBatch.expiryDate = transactionBatch.expiryDate;
-        itemBatch.batch = transactionBatch.batch;
-        itemBatch.costPrice = transactionBatch.costPrice;
-        itemBatch.sellPrice = transactionBatch.sellPrice;
-        database.save('ItemBatch', itemBatch);
-      });
+    const externalInvoice = this.isExternalSI;
```

Comment:
renamed to `isExternalInvoice` and `isIncomingInvoice` I think it goes without saying and references that repetitive string validations (in both functions) shouldn't be happening on every iteration when they can happen once and rammed.  Good practice re booleans though, I missed that one cheers.